src/preload.py

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/python3
+
+import json
+import os
+
+from contextlib import contextmanager
+from distutils.version import LooseVersion
+from math import ceil, floor
+from requests import get
+from sh import (
+    btrfs,
+    docker,
+    dockerd,
+    e2fsck,
+    kill,
+    losetup,
+    mount,
+    parted,
+    resize2fs,
+    umount,
+    inotifywait,
+    ErrorReturnCode,
+)
+from shutil import copyfile, rmtree
+from sys import exit
+from tempfile import mkdtemp
+
+os.environ["LANG"] = "C"
+
+API_TOKEN = os.environ["API_TOKEN"]
+API_KEY = os.environ["API_KEY"]
+APP_ID = os.environ["APP_ID"]
+
+API_HOST = os.environ["API_HOST"] or "https://api.resin.io"
+REGISTRY_HOST = os.environ["REGISTRY_HOST"] or "registry2.resin.io"
+
+
+def human_size(size, precision=2):
+    suffixes = ['', 'Ki', 'Mi', 'Gi', 'Ti']
+    idx = 0
+    while idx < 4 and size >= 1024:
+        idx += 1
+        size /= 1024
+    result = "{:.{}f}".format(size, precision).rstrip("0").rstrip(".")
+    return "{} {}B".format(result, suffixes[idx])
+
+
+def log(*args):
```

Comment:
Question, could standard `logging` be used for logs, to achieve the same effect? I feel it would be nicer (even though this current one is lightweight)